Mathlib/SetTheory/Surreal/Division.lean

```diff
@@ -0,0 +1,757 @@
+proved for positive numbers `x`. Once we have defined the inverse for positive `x`, it is extended
+in the obvious way to negative numbers.
+
+The desired final results are (ii) and (iv) which state that `x⁻¹` is a number and is a
+multiplicative inverse, respectively.
+
+In our proof, lemmas (i) through (iv) are proved for positive numbers in the section labeled
+`Positive` by a series of preliminary lemmas, culminating in the lemma `onag_1_10` which
+proves (ii) and (iv) by a joint induction over the two statements. Lemmas (i) and (iii) are used
+only within the proof of `onag_1_10` and are not needed as induction hypotheses.
+
+We do not use the notation `x⁻¹` in the proof of lemmas (i) to (iv); we use `x.inv'` instead
+since that is the definition for positive `x`. We only use `x⁻¹` at the very end when we define
+the inverse for all nonzero surreals, both positive and negative.
+
+## Differences with ONAG
+
+Conway starts by replacing `x` with an equal game which I have called its `normalization`.
+However, the normalization is not actually needed until parts (iii) and (iv) of the proof,
+so we do not use it until those parts.
+
+For convenience, we split (iii) into 2 lemmas, the Left and Right statements.
+
+The proof of (iii) in ONAG has a minor gap. It does not work for the Left option `x' = 0`. However,
+in that case, (iii) reduces to (i). It is treated as a separate case in our proof.
+
+## Notation
+
+Throughout the proof of lemmas (i) to (iv), we use the same notation as in ONAG:
+
+`x` : the positive number whose inverse we are constructing
+`x'` : a positive option of `x`, indexed by `i`
+`y` : the purported inverse of `x`, as defined by `x.inv'`
+`y'` : an option of `y`, indexed by `i'`
+`y''` : `(1 + (x' - x) * y') / x'` , the form of any option of `y` besides 0, indexed by `i''`
+
+## References
+
+* [Conway, *On numbers and games*][Conway2001]
+
+-/
+open SetTheory
+open PGame
+open Equiv
+
+variable {x : PGame} (x_num : x.Numeric)
+
+namespace Surreal.Division
+section Positive
+
+variable (x_pos : 0 < x)
+
+/-! ### A few random positivity lemmas -/
+
+/-- This applies to an arbitrary `LinearCommOrderedRing` and probably should go elsewhere. -/
+lemma inv_pos_of_pos {x y : Surreal} (h_inv: x * y = 1) (h : 0 < x) : 0 < y := by
+  apply (pos_iff_pos_of_mul_pos _).mp
+  on_goal 2 => exact x
+  · exact h
+  · simp only [h_inv, zero_lt_one]
+
+lemma right_pos_of_pos : ∀ j, 0 < x.moveRight j := by
+  apply le_of_lt at x_pos
+  rw [le_iff_forall_lt] at x_pos
+  simp only [zero_leftMoves, IsEmpty.forall_iff, true_and] at x_pos
+  · exact x_pos
+  · exact numeric_zero
+  · exact x_num
+
+/-! ### Normalization of a positive numeric game -/
+
+/-- If `x` is a positive numeric game, then keeping only the positive Left options then inserting
+  0 as a Left option, results in an equal game. This game is called the normalization of `x`.
+-/
+def normalization (x : PGame) : PGame :=
```

Comment:
I've forgotten my ONAG: if this is useful beyond setting up the field structure, perhaps this could preemptively be moved to its own file? Otherwise fine as is.

Reply:
I don't see this normalization used in ONAG beyond setting up the field structure.